src/handshakes/abstract-handshake.ts

```diff
@@ -89,13 +72,12 @@ export abstract class AbstractHandshake {
     return ciphertext
   }
 
-  protected decrypt (k: bytes32, n: uint64, ad: bytes, ciphertext: bytes): {plaintext: bytes, valid: boolean} {
-    if (n > MAX_NONCE) {
+  protected decrypt (k: bytes32, n: Nonce, ad: bytes, ciphertext: bytes): {plaintext: bytes, valid: boolean} {
+    if (n.getUint64() > MAX_NONCE) {
       throw new Error(ERR_MAX_NONCE)
     }
```

Comment:
This can be moved into a method on `Nonce`.
```ts
assertNonceInBounds(): void {
  if (this.n > MAX_NONCE) {
    throw new Error(ERR_MAX_NONCE)
  }
}
```